thunder/torch/__init__.py

```diff
@@ -5668,7 +5695,7 @@ def autograd_function_apply(
     args_tensor_mask: Sequence[bool] | None,
     non_differentiable_idx: Sequence[int] | None = None,
 ) -> TensorProxy | tuple[TensorProxy, ...]:
-    result, saved_for_backward = fwd(None, *args)
+    result, saved_for_backward = call_higher_order_function_and_consider_outer_autograd_setting(fwd)(None, *args)
```

Comment:
Unfortunately, the jit lookaside for `torch.ops.higher_order.autograd_function_apply` doesn't use this implementation and the same change needs to be duplicated in
https://github.com/Lightning-AI/lightning-thunder/blob/76f7e81f7dbba0c47e535781b79e8850cce48efb/thunder/core/jit_ext.py#L776

Reply:
Have filed https://github.com/Lightning-AI/lightning-thunder/issues/1612 for the same.

Reply:
Of course!